From jirislaby@gmail.com  Thu Apr 16 21:28:57 2009
From: Jiri Slaby <jirislaby@gmail.com>
Date: Thu, 26 Mar 2009 09:34:08 +0100
Subject: staging: agnx, remove memlens from probe
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Jiri Slaby <jirislaby@gmail.com>
Message-ID: <1238056453-25653-3-git-send-email-jirislaby@gmail.com>


Pass 0 to pci_iomap instead. It will cope with that per se.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/agnx/pci.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/staging/agnx/pci.c
+++ b/drivers/staging/agnx/pci.c
@@ -455,7 +455,6 @@ static int __devinit agnx_pci_probe(stru
 {
 	struct ieee80211_hw *dev;
 	struct agnx_priv *priv;
-	u32 mem_len0, mem_len1;
 	int err;
 	DECLARE_MAC_BUF(mac);
 
@@ -465,9 +464,6 @@ static int __devinit agnx_pci_probe(stru
 		return err;
 	}
 
-	mem_len0 = pci_resource_len(pdev, 0);
-	mem_len1 = pci_resource_len(pdev, 1);
-
 	err = pci_request_regions(pdev, "agnx-pci");
 	if (err) {
 		dev_err(&pdev->dev, "can't reserve PCI resources\n");
@@ -496,13 +492,13 @@ static int __devinit agnx_pci_probe(stru
 	spin_lock_init(&priv->lock);
 	priv->init_status = AGNX_UNINIT;
 
-	priv->ctl = pci_iomap(pdev, 0, mem_len0);
+	priv->ctl = pci_iomap(pdev, 0, 0);
 /*	dev_dbg(&pdev->dev, "MEM1 mapped address is 0x%p\n", priv->ctl); */
 	if (!priv->ctl) {
 		dev_err(&pdev->dev, "can't map device memory\n");
 		goto err_free_dev;
 	}
-	priv->data = pci_iomap(pdev, 1, mem_len1);
+	priv->data = pci_iomap(pdev, 1, 0);
 	if (!priv->data) {
 		dev_err(&pdev->dev, "can't map device memory\n");
 		goto err_iounmap2;
